ur_robot_driver/src/ros/hardware_interface.cpp

```diff
@@ -705,6 +711,12 @@ void HardwareInterface::publishToolData()
       tool_data_pub_->msg_.tool_output_voltage = tool_output_voltage_;
       tool_data_pub_->msg_.tool_current = tool_output_current_;
       tool_data_pub_->msg_.tool_temperature = tool_temperature_;
+      tool_data_pub_->msg_.output_double_register_24 = output_double_register_[0];
+      tool_data_pub_->msg_.output_double_register_25 = output_double_register_[1];
+      tool_data_pub_->msg_.output_double_register_26 = output_double_register_[2];
+      tool_data_pub_->msg_.output_double_register_27 = output_double_register_[3];
+      tool_data_pub_->msg_.output_double_register_28 = output_double_register_[4];
+      tool_data_pub_->msg_.output_double_register_29 = output_double_register_[5];
```

Comment:
This does obviously not build, as the message doesn't contain such a field.

Reply:
I would also suggest to include the effort data in the appropriate message (ie: `JointState`). This can be accomplished by updating the appropriate member variable of the hw interface (which in the end gets read by the `joint_state_controller`).

Reply:
I think that was the plan initially.

Reply:
@gtatiya: you should remove these lines.